addons/fleet/models/fleet_vehicle.py

```diff
@@ -259,11 +258,20 @@ def _search_get_overdue_contract_reminder(self, operator, value):
         else:
             search_operator = 'not in'
         today = fields.Date.context_today(self)
-        res_ids = self.env['fleet.vehicle.log.contract'].search([
+        # get the id of vehicles that have overdue contracts
+        expired_ids = set(self.env['fleet.vehicle.log.contract'].search([
             ('expiration_date', '!=', False),
             ('expiration_date', '<', today),
             ('state', 'in', ['open', 'expired'])
-        ]).mapped('vehicle_id').ids
+        ]).mapped('vehicle_id').ids)
```

Comment:
![image](https://github.com/user-attachments/assets/c5901ea7-1c52-4ae1-a343-cabe75da313d)

No need to have set.. 

https://www.odoo.com/documentation/17.0/developer/reference/backend/orm.html?highlight=orm#map

also mapped is old way of doing it.. we don't just modify code just for the sake to remove mapped... but if you are already changing the line, if we could already as well remove mapped

`Since V13, multi-relational field access is supported and works like a mapped call
records.partner_id  # == records.mapped('partner_id')
`

```
expired_ids = self.env['fleet.vehicle.log.contract'].search([
            ('expiration_date', '!=', False),
            ('expiration_date', '<', today),
            ('state', 'in', ['open', 'expired'])
        ]).vehicle_id
```